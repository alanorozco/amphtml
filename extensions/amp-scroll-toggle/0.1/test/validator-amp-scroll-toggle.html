<!--
  Copyright 2019 The AMP HTML Authors. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS-IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the license.
-->
<!--
  Test Description:
  Tests for the amp-scroll-toggle tag.
-->
<!doctype html>
<html âš¡>
<head>
  <meta charset="utf-8">
  <title>amp-scroll-toggle example</title>
  <link rel="canonical" href="amps.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async custom-element="amp-scroll-toggle" src="https://cdn.ampproject.org/v0/amp-scroll-toggle-0.1.js"></script>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
</head>
<body>

  <!-- Example target reference. -->
  <header id="a">My website</header>

  <!-- Valid: layout and target -->
  <amp-scroll-toggle layout="nodisplay" target="a"></amp-scroll-toggle>

  <!-- Valid: layout, target and position=top -->
  <amp-scroll-toggle layout="nodisplay" target="a" position="top">
  </amp-scroll-toggle>

  <!-- Valid: layout, target and position=bottom -->
  <amp-scroll-toggle layout="nodisplay" target="a" position="bottom">
  </amp-scroll-toggle>

  <!-- Invalid position -->
  <amp-scroll-toggle layout="nodisplay" target="a" position="left">
  </amp-scroll-toggle>

  <!-- Invalid: target with invalid chars -->
  <amp-scroll-toggle layout="nodisplay" target="$"></amp-scroll-toggle>

  <!-- Invalid: empty target -->
  <amp-scroll-toggle layout="nodisplay" target></amp-scroll-toggle>

  <!-- Invalid: no target -->
  <amp-scroll-toggle layout="nodisplay"></amp-scroll-toggle>

  <!-- Invalid layout -->
  <amp-scroll-toggle layout="responsive" target="a"></amp-scroll-toggle>
</body>
</html>
