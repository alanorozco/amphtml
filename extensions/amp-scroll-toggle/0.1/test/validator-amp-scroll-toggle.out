FAIL
|  <!--
|    Copyright 2019 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    Tests for the amp-scroll-toggle tag.
|  -->
|  <!doctype html>
|  <html âš¡>
|  <head>
|    <meta charset="utf-8">
|    <title>amp-scroll-toggle example</title>
|    <link rel="canonical" href="amps.html">
|    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|    <script async custom-element="amp-scroll-toggle" src="https://cdn.ampproject.org/v0/amp-scroll-toggle-0.1.js"></script>
|    <script async src="https://cdn.ampproject.org/v0.js"></script>
|  </head>
|  <body>
|
|    <!-- Valid: layout and target -->
|    <amp-scroll-toggle layout="nodisplay" target="a"></amp-scroll-toggle>
|
|    <!-- Valid: layout, target and position=top -->
|    <amp-scroll-toggle layout="nodisplay" target="a" position="top">
|    </amp-scroll-toggle>
|
|    <!-- Valid: layout, target and position=bottom -->
|    <amp-scroll-toggle layout="nodisplay" target="a" position="bottom">
|    </amp-scroll-toggle>
|
|    <!-- Invalid position -->
|    <amp-scroll-toggle layout="nodisplay" target="a" position="left">
>>   ^~~~~~~~~
amp-scroll-toggle/0.1/test/validator-amp-scroll-toggle.html:45:2 The attribute 'position' in tag 'amp-scroll-toggle' is set to the invalid value 'left'. (see https://www.ampproject.org/docs/reference/components/amp-scroll-toggle) [DISALLOWED_HTML]
|    </amp-scroll-toggle>
|
|    <!-- Invalid: target with invalid chars -->
|    <amp-scroll-toggle layout="nodisplay" target="$"></amp-scroll-toggle>
>>   ^~~~~~~~~
amp-scroll-toggle/0.1/test/validator-amp-scroll-toggle.html:49:2 The attribute 'target' in tag 'amp-scroll-toggle' is set to the invalid value '$'. (see https://www.ampproject.org/docs/reference/components/amp-scroll-toggle) [DISALLOWED_HTML]
|
|    <!-- Invalid: empty target -->
|    <amp-scroll-toggle layout="nodisplay" target></amp-scroll-toggle>
>>   ^~~~~~~~~
amp-scroll-toggle/0.1/test/validator-amp-scroll-toggle.html:52:2 The attribute 'target' in tag 'amp-scroll-toggle' is set to the invalid value ''. (see https://www.ampproject.org/docs/reference/components/amp-scroll-toggle) [DISALLOWED_HTML]
|
|    <!-- Invalid: no target -->
|    <amp-scroll-toggle layout="nodisplay"></amp-scroll-toggle>
>>   ^~~~~~~~~
amp-scroll-toggle/0.1/test/validator-amp-scroll-toggle.html:55:2 The mandatory attribute 'target' is missing in tag 'amp-scroll-toggle'. (see https://www.ampproject.org/docs/reference/components/amp-scroll-toggle) [AMP_TAG_PROBLEM]
|
|    <!-- Invalid layout -->
|    <amp-scroll-toggle layout="responsive" target="a"></amp-scroll-toggle>
>>   ^~~~~~~~~
amp-scroll-toggle/0.1/test/validator-amp-scroll-toggle.html:58:2 The specified layout 'RESPONSIVE' is not supported by tag 'amp-scroll-toggle'. (see https://www.ampproject.org/docs/reference/components/amp-scroll-toggle) [AMP_LAYOUT_PROBLEM]
|  </body>
|  </html>